### ISSUE TITLE
Render templates during build

### ISSUE THREAD

**Issue by zdarovka** [frhagn/Typewriter#258]
Is it possible to set TypeWriter to only render templates during build? Rendering templates on saving file is slowing VisualStudio perfomance as it needs to render many files.

**Comment by RudeySH** [comment]
But doing this would stop you from being able to modify scripts while debugging.

However, rendering templates on build would be useful nonetheless. Currently, when checking out a project for the first time that uses Typewriter, simply building the solution no longer works. It requires me to manually locate a *.tst file and press CTRL+S before I can continue working with the project.

**Comment by agustinbcu01** [comment]
Yes that can ensure we have the last generate code, avoiding some merge issues. For example if A modify the tst file and B create a new .cs file, we can fall in old code generation issue merging PR without conflicts, and an automated CI build process can fail.

**Comment by gregveres** [comment]
@RudeySH There is a render all templates option that is useful in that situation. I believe you can right click on the project and select render all templates.

@zdarovka I think there is a better solution. I was coming over here to request a different feature. I have now built up my project so that there are many many files being generated from the templates and I am using webpack's dev server to pack them up to be used during local testing. 

When I am working exclusively in C# on unit tests, what happens is that each time I save one of my C# files, it causes the template to be re-rendered and that causes the TS file to be written out, which causes the TS complier to run and webpack to repack the file. But all this work happens when nothing changed in the generated TS file. And what is worse is that the TS file that is getting generated is like a header file being included in many other TS files, causing a big chain reaction. 

I think a better solution to all of this would be to have the template rendered to a temp file that can be compared against the already generated file. If the new file is different, then replace the original. If the new file is the same, then replace it. This would still cause the templates to be rendered, but the chain would be broken there. 

I am going to open a new issue requesting that.

**Comment by Cubelaster** [comment]
+ For render on build. Would be really useful when running CI and auto build + deploy.

**Comment by Gambero81** [comment]
this is a must feature to manage automated CI build process.

@frhagn can be scheduled for implementation?

Thanks
